ui/components/app/transaction-decoding/transaction-decoding.component.js

```diff
@@ -0,0 +1,224 @@
+import React, { useContext, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import inspect from 'browser-util-inspect';
+import { forAddress } from '@truffle/decoder';
+import { useSelector } from 'react-redux';
+import * as Codec from '@truffle/codec';
+import Spinner from '../../ui/spinner';
+import ErrorMessage from '../../ui/error-message';
+import fetchWithCache from '../../../helpers/utils/fetch-with-cache';
+import { getSelectedAccount, getCurrentChainId } from '../../../selectors';
+import { hexToDecimal } from '../../../helpers/utils/conversions.util';
+import { I18nContext } from '../../../contexts/i18n';
+import { toChecksumHexAddress } from '../../../../shared/modules/hexstring-utils';
+import { transformTxDecoding } from './transaction-decoding.util';
+import {
+  FETCH_PROJECT_INFO_URI,
+  FETCH_SUPPORTED_NETWORKS_URI,
+} from './constants';
+
+import Address from './components/decoding/address';
+import CopyRawData from './components/ui/copy-raw-data';
+
+export default function TransactionDecoding({ to = '', inputData: data = '' }) {
+  const t = useContext(I18nContext);
+  const [tx, setTx] = useState([]);
+  const { address: from } = useSelector(getSelectedAccount);
+  const network = hexToDecimal(useSelector(getCurrentChainId));
+
+  const [loading, setLoading] = useState(false);
+  const [hasError, setError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    (async () => {
+      setLoading(true);
+      try {
+        const networks = await fetchWithCache(FETCH_SUPPORTED_NETWORKS_URI, {
+          method: 'GET',
+        });
+
+        if (
+          !networks.some(
+            (n) => n.active && Number(n.chainId) === Number(network),
+          )
+        ) {
+          throw new Error(
+            t('transactionDecodingUnsupportedNetworkError', [network]),
+          );
+        }
+
+        const requestUrl = `${FETCH_PROJECT_INFO_URI}?${new URLSearchParams({
+          to,
+          'network-id': network,
+        })}`;
+
+        const response = await fetchWithCache(requestUrl, { method: 'GET' });
+
+        const { info: projectInfo } = response;
+
+        // creating instance of the truffle decoder
+        const decoder = await forAddress(to, {
+          provider: global.ethereumProvider,
+          projectInfo,
+        });
+
+        // decode tx input data
+        const decoding = await decoder.decodeTransaction({
+          from,
+          to,
+          input: data,
+          blockNumber: null,
+        });
+
+        // fake await
+        await new Promise((resolve) => {
```

Comment:
Why is this 500ms pause needed?

Reply:
that's a good question

Reply:
we need to remove it, we had it temporarily as sometimes we get response faster than expected where we can not show the loading spinner ! but we can totally remove it.

Reply:
removed with 5ffefd449